src/ImageIO/CGImageDestination.cs

```diff
@@ -470,5 +495,15 @@ public bool CopyImageSource (CGImageSource image, CGImageDestinationOptions opti
 					o.Dispose ();
 			}
 		}
+
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[DllImport (Constants.ImageIOLibrary)]
+		static extern void CGImageDestinationAddAuxiliaryDataInfo (IntPtr /* CGImageDestinationRef* */ idst, IntPtr /* CFStringRef* */ auxiliaryImageDataType, IntPtr /* CFDictionaryRef* */ auxiliaryDataInfoDictionary);
+
+		public void AddAuxiliaryDataInfo (CGImageDestination dest, CGImageAuxiliaryDataType auxiliaryImageDataType, CGImageAuxiliaryDataInfo auxiliaryDataInfo)
+		{
+			var dict = auxiliaryDataInfo == null ? null : auxiliaryDataInfo.ToDictionary ();
```

Comment:
The GC could collect the `NSDictionary` while running the next native call. It's also not disposed manually (even if it should not be big, it could be noticeable when run into a loop).

```
using (var dict =  auxiliaryDataInfo?.ToDictionary ()) {
   ...
}
```

would solve both.